pkg/root/trusted_root.go

```diff
@@ -256,3 +259,101 @@ func NewTrustedRootProtobuf(rootJSON []byte) (*prototrustroot.TrustedRoot, error
 	}
 	return pbTrustedRoot, nil
 }
+
+// FetchTrustedRoot fetches the Sigstore trusted root from TUF and returns it.
+func FetchTrustedRoot() (*TrustedRoot, error) {
+	opts, err := tuf.DefaultOptions()
+	if err != nil {
+		return nil, err
+	}
+	client, err := tuf.New(opts)
+	if err != nil {
+		return nil, err
+	}
+	return GetTrustedRoot(client)
+}
+
+// GetTrustedRoot returns the trusted root
+func GetTrustedRoot(c *tuf.Client) (*TrustedRoot, error) {
+	jsonBytes, err := c.GetTarget("trusted_root.json")
+	if err != nil {
+		return nil, err
+	}
+	return NewTrustedRootFromJSON(jsonBytes)
+}
+
+// LiveTrustedRoot is a wrapper around TrustedRoot that periodically
+// refreshes the trusted root from TUF. This is needed for long-running
+// processes to ensure that the trusted root does not expire.
+type LiveTrustedRoot struct {
+	*TrustedRoot
+	mu sync.RWMutex
+}
+
+// NewLiveTrustedRoot returns a LiveTrustedRoot that will periodically
+// refresh the trusted root from TUF.
+func NewLiveTrustedRoot(opts *tuf.Options) (*LiveTrustedRoot, error) {
```

Comment:
Maybe for later, do you think this should be built into the default TrustedRoot rather than maintain a separate one? It shouldn't impact performance beyond having a background thread.

Reply:
That's a fair assessment! The `LiveTrustedRoot` was kind of a shower thought I had as we were about to open-source it, but if others agree, we can definitely make this the default to reduce API surface and ensure that the default client is compatible with long-running processes. I agree that it doesn't affect performance too much. Just a mutex hit and background thread.

Reply:
Just realized there's a bit of a foot-gun with this func... If a long-running process calls `NewLiveTrustedRoot` for every verification (if the author doesn't realize it auto-updates), it will spawn goroutines for each invocation, so the clients would not get garbage collected and would gradually fill up memory and threads.